src/sources/kubernetes_logs/mod.rs

```diff
@@ -217,20 +224,26 @@ struct Source {
 }
 
 impl Source {
-    fn new(
+    async fn new(
         config: &Config,
         globals: &GlobalOptions,
         key: &ComponentKey,
-        proxy: &ProxyConfig,
+        _proxy: &ProxyConfig,
```

Comment:
TODO: I convinced @jszwedko that this could be removed, if there are no other objections from the reviewers I'll go from ignoring to actually removing...

The tl;dr is that a `kubeconfig` or config for the kube client already has ways to set the address for the API server, which would be used for proxying if necessary, and that proxying doesn't really make sense unless the client is run outside of the cluster - which isn't a supported mode of the source.